rdr_service/dao/genomics_dao.py

```diff
@@ -3040,6 +3040,46 @@ def get_report_state_from_wf_state(wf_state):
                 return value
         return None
 
+    def get_hdr_result_positive_no_appointment(self) -> List[str]:
+        """Returns participants with hdr positive result over 14 days ago and no appointment scheduled/completed"""
+        result_before = clock.CLOCK.now() - timedelta(days=14)
+
+        max_event_authored_time_subquery = sqlalchemy.orm.Query(
+            functions.max(GenomicAppointmentEvent.event_authored_time).label(
+                'max_event_authored_time'
+            )
+        ).filter(
+            GenomicAppointmentEvent.event_type.notlike('%note_available'),
+            GenomicAppointmentEvent.module_type == 'hdr'
+        ).group_by(
+            GenomicAppointmentEvent.participant_id,
+        ).subquery()
+
+        with self.session() as session:
+            pids = session.query(
+                GenomicMemberReportState.participant_id
+            ).outerjoin(
+                GenomicAppointmentEvent,
+                and_(
+                    GenomicAppointmentEvent.participant_id == GenomicMemberReportState.participant_id,
+                    GenomicAppointmentEvent.module_type == 'hdr'
+                )
+            ).outerjoin(
+                GenomicGCROutreachEscalationNotified,
+                GenomicMemberReportState.participant_id == GenomicGCROutreachEscalationNotified.participant_id
+            ).filter(
+                GenomicGCROutreachEscalationNotified.participant_id.is_(None),
+                GenomicMemberReportState.genomic_report_state == GenomicReportState.HDR_RPT_POSITIVE,
+                GenomicMemberReportState.event_authored_time < result_before,
+                or_(GenomicAppointmentEvent.event_type.notin_(('appointment_completed', 'appointment_scheduled',
+                                                               'appointment_updated')),
+                    GenomicAppointmentEvent.event_type.is_(None)),
+                or_(GenomicAppointmentEvent.event_authored_time ==
+                    max_event_authored_time_subquery.c.max_event_authored_time,
+                    GenomicAppointmentEvent.event_authored_time.is_(None))
+            ).distinct().all()
+        return [pid[0] for pid in pids]
```

Comment:
This is aa little outside of the realm of what we utilize the daos for, can you do this list-casting in the method where you are utilizing this query?

Reply:
This has been moved